python/pyspark/ml/pipeline.py

```diff
@@ -89,9 +87,9 @@ def setParams(self, stages=None):
         setParams(self, stages=None)
         Sets params for Pipeline.
         """
-        if stages is None:
-            stages = []
         kwargs = self.setParams._input_kwargs
+        if stages is None:
+            kwargs['stages'] = []
```

Comment:
Well if you actually call `setParams` with `stages=None` like the function shows as the default, it seems like it should work, but fails when `_fit` is called.  Maybe people would know not to do this and it's not really a problem?